tests/integration/test_pip.py

```diff
@@ -89,3 +91,89 @@ def test_pip_packages(
+    [
+        # Test case checks fetching pip dependencies, generating environment vars file,
+        # building image with all prepared prerequisites and testing if pip packages are present
+        # in built image
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/pip-e2e-test.git",
+                ref="03939ea78542c7cbfe7a86c5a4133612813e9ec6",
+                packages=(
+                    {
+                        "type": "pip",
+                        "requirements_files": ["requirements.txt"],
+                        "requirements_build_files": ["requirements-build.txt"],
+                    },
+                ),
+                check_vendor_checksums=False,
+                expected_exit_code=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+            ["python3", "/opt/test_package_cachi2"],
+            id="pip_e2e_test",
+        ),
+    ],
+)
+def test_e2e_pip(
+    test_params: utils.TestParameters,
+    check_cmd: List[str],
+    cachi2_image: utils.ContainerImage,
+    tmpdir: Path,
+    test_data_dir: Path,
+    request: pytest.FixtureRequest,
+):
+    """
+    End to end test for gomod.
+
+    :param test_params: Test case arguments
+    :param tmpdir: Temp directory for pytest
+    """
+    test_case = request.node.callspec.id
+
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, f"{test_case}-source", tmpdir
+    )
+
+    output_folder = utils.fetch_deps_and_check_output(
+        tmpdir, test_case, test_params, source_folder, test_data_dir, cachi2_image
+    )
+
+    log.info("Create cachi2.env file")
+    env_vars_file = os.path.join(tmpdir, "cachi2.env")
+    cmd = [
+        "generate-env",
+        output_folder,
+        "--output",
+        env_vars_file,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Env var file creation failed. output-cmd: {output}"
+
+    log.info("Inject project files")
+    cmd = [
+        "inject-files",
+        output_folder,
+        "--for-output-dir",
+        os.path.join("/tmp", f"{test_case}-output"),
+    ]
+    (output, exit_code) = cachi2_image.run_cmd_on_image(cmd, tmpdir)
+    assert exit_code == 0, f"Injecting project files failed. output-cmd: {output}"
+
+    log.info("Build container image with all prerequisites retrieved in previous steps")
+    container_folder = os.path.join(test_data_dir, test_case, "container")
+
+    with utils.build_image(
+        tmpdir, os.path.join(container_folder, "Containerfile"), test_case
+    ) as test_image:
+        log.info(f"Run command {check_cmd} on built image {test_image.repository}")
+        (output, exit_code) = test_image.run_cmd_on_image(check_cmd, tmpdir)
+
+        assert exit_code == 0, f"{check_cmd} command failed, Output: {output}"
```

Comment:
Would it make sense to have a utils method for running the e2e part of the test, to avoid duplication between this and the gomod e2e tests? 

Similar to the `fetch_deps_and_check_output method`, maybe `build_image_and_check_cmd`?

Reply:
yeah sure!